### PR TITLE
[SPARK-43380][SQL] Revert `Fix Avro data type conversion issues`

### PR DESCRIPTION
**Pull request by zeruibao**
### What changes were proposed in this pull request?
Revert my last PR https://github.com/apache/spark/pull/41052 that causes AVRO read performance regression since I change the code structure.

### Why are the changes needed?
Remove performance regression

### How was this patch tested?
Unit test

### CODE REVIEW

**Comment by gengliangwang** [comment]
@zeruibao Just double confirm: is this a clean revert?

File: connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala [hunk 1]
```diff
@@ -423,6 +334,17 @@ private[sql] class AvroDeserializer(
...
+      case (INT, _: YearMonthIntervalType) => (updater, ordinal, value) =>
+        updater.setInt(ordinal, value.asInstanceOf[Int])
+
+      case (LONG, _: DayTimeIntervalType) => (updater, ordinal, value) =>
+        updater.setLong(ordinal, value.asInstanceOf[Long])
+
+      case (LONG, _: DecimalType) => (updater, ordinal, value) =>
+        val d = avroType.getLogicalType.asInstanceOf[CustomDecimal]
+        updater.setDecimal(ordinal, Decimal(value.asInstanceOf[Long], d.precision, d.scale))
+
```
Comment (zeruibao) [inline connector/avro/src/main/scala/org/apache/spark/sql/avro/AvroDeserializer.scala:347]: Only change is here @gengliangwang

**Comment by zeruibao** [comment]
@gengliangwang It's a clean revert with the support for `CustomDecimal`

**Review by gengliangwang — APPROVED** [review]

File: docs/sql-error-conditions.md [hunk 2]
```diff
@@ -93,18 +93,6 @@ Invalid as-of join.
...
-SQLSTATE: none assigned
-
-Cannot convert Avro `<avroPath>` to SQL `<sqlPath>` because the original encoded data type is `<avroType>`, however you're trying to read the field as `<sqlType>`, which would lead to an incorrect answer. To allow reading this field, enable the SQL configuration: `<key>`.
-
-### AVRO_LOWER_PRECISION
-
-SQLSTATE: none assigned
-
-Cannot convert Avro `<avroPath>` to SQL `<sqlPath>` because the original encoded data type is `<avroType>`, however you're trying to read the field as `<sqlType>`, which leads to data being read as null. Please provide a wider decimal type to get the correct result. To allow reading null to this field, enable the SQL configuration: `<key>`.
-
```
Comment (zeruibao) [inline docs/sql-error-conditions.md:107]: @gengliangwang This is the new change. Just remove it from the doc.

Comment (gengliangwang) [reply docs/sql-error-conditions.md:107]: What about the migration guide doc changes? https://github.com/apache/spark/pull/41052/files#diff-b6c16c7a1a4e60bacda4674b52f9eecde3c8a86f281fb3ef21052267f94c1f01R29

Comment (zeruibao) [reply docs/sql-error-conditions.md:107]: Also revert it as well.

**Comment by gengliangwang** [comment]
Thanks, merging to master/branch-3.5

**Pull request closed without merging**